### PR TITLE
Controller and View changes for Add trade page

### PR DESCRIPTION
**Pull request by chandraAndem**

### CODE REVIEW

File: app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessTradeNameController.scala [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import uk.gov.hmrc.play.language.LanguageUtils
+
+import scala.concurrent.{ExecutionContext, Future}
+
+
+@Singleton
+class BusinessTradeNameController @Inject()(mcc: MessagesControllerComponents,
+                                            incomeTaxSubscriptionConnector: IncomeTaxSubscriptionConnector,
+                                            authService: AuthService,
+                                            val languageUtils: LanguageUtils)
```
Comment (AlexRimmerHMRC) [inline app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessTradeNameController.scala:43]: languageUtils is not required

File: app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessTradeNameController.scala [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+                                           (implicit val ec: ExecutionContext, val appConfig: AppConfig) extends FrontendController(mcc)
+  with I18nSupport with ImplicitDateFormatter {
```
Comment (AlexRimmerHMRC) [inline app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessTradeNameController.scala:45]: ImplicitDateFormatter is not used

File: app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessTradeNameController.scala [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+
+  def submit(): Action[AnyContent] = Action.async { implicit request =>
+    authService.authorised() {
+      businessTradeNameValidationForm.bindFromRequest.fold(
+        formWithErrors =>
+          Future.successful(BadRequest(view(formWithErrors))),
+        businessTradeName =>
+          incomeTaxSubscriptionConnector.saveSelfEmployments(BusinessTradeNameController.businessTradeNameKey, businessTradeName) map (_ =>
+            Redirect(uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.controllers.routes.BusinessStartDateController.show())
```
Comment (AlexRimmerHMRC) [inline app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessTradeNameController.scala:73]:
```suggestion
            Redirect(uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.controllers.routes.BusinessTradeNameController.show())
```
Can this redirect to it's own show method?

File: app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessTradeNameController.scala [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+            )
+      )
+    }
+  }
+
+  def backUrl(): String =
+    appConfig.incomeTaxSubscriptionFrontendBaseUrl + "/business/address"
```
Comment (AlexRimmerHMRC) [inline app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessTradeNameController.scala:80]: Back url should go to the business name page

File: conf/messages.cy [hunk 5]
```diff
@@ -35,3 +35,15 @@ business.name.line_1                                            = Dyma enw’r b
...
 error.business_name.max_length                                  = Mae’n rhaid i enw’r busnes fod yn 160 o gymeriadau neu lai
 error.business_name.invalid_character                           = Ni all enw’r busnes gynnwys !, “ “ , * na ?
+
+# Business trade name
+business.trade_name.title                                       = Beth yw masnach eich busnes?
+business.trade_name.heading                                     = Beth yw masnach eich busnes?
+business.trade_name.hint                                        = Er enghraifft: plymio, gwaith trydanol, ymgynghori
+
+# Business trade name errors
+error.business_trade_name.empty                                 = Enter the trade of your business
```
Comment (AlexRimmerHMRC) [inline conf/messages.cy:45]: This isn't welsh

File: app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/models/BusinessTradeNameModel.scala [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+ */
+
+package uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.models
+
+import play.api.libs.json.{Json, OFormat}
+
+case class BusinessTradeNameModel(businessTradeName: String)
+
+object BusinessTradeNameModel {
+  implicit val format: OFormat[BusinessTradeNameModel] = Json.format[BusinessTradeNameModel]
```
Comment (AlexRimmerHMRC) [inline app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/models/BusinessTradeNameModel.scala:24]: Could you add a model spec?

**Comment by AlexRimmerHMRC** [comment]
The GetAllSelfEmploymentModel needs updating with the new answer in it

File: app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessTradeNameController.scala (hunk 1 above)
Comment (chandraAndem) [reply app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessTradeNameController.scala:43]: Removed

File: app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessTradeNameController.scala (hunk 2 above)
Comment (chandraAndem) [reply app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessTradeNameController.scala:45]: Removed

File: app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessTradeNameController.scala (hunk 4 above)
Comment (chandraAndem) [reply app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessTradeNameController.scala:80]: Done

File: conf/messages.cy (hunk 5 above)
Comment (chandraAndem) [reply conf/messages.cy:45]: Updated

File: app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/models/BusinessTradeNameModel.scala (hunk 6 above)
Comment (chandraAndem) [reply app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/models/BusinessTradeNameModel.scala:24]: Added

**Pull request merged**